### PR TITLE
[#10031] Cancel causes permissions to be shown incorrectly in instructor course edit page

### LINKED ISSUE #10031: Cancel causes permissions to be shown incorrectly in instructor course edit page
**Environment**:  `master` branch at commit 7e9e884db042a091345b43f09749f3167149d446

**Steps to reproduce**
1. Add instructor with custom permissions
2. Click edit
3. Click cancel
4. Click edit again

**Expected behaviour**
The correct custom permissions for the instructor should be shown 

**Actual behaviour**
All checkboxes are ticked when edit is clicked again. Section and session level permissions are also not shown.

**Additional info**
Instructor with custom permissions, should show correct permissions:
before

After clicking cancel, wrong permissions shown:
after

### PR DESCRIPTION
**Pull request by jtankw3**
Fixes #10031

**Outline of Solution**
- Add extra field to store originalPanel, set editPanel back to originalPanel on cancel so that permissions are preserved

### CODE REVIEW

**Comment by jtankw3** [comment]
Here is a GIF to show the permissions are preserved. 

ezgif-3-816bfe79da27

File: src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts [hunk 1]
```diff
@@ -336,7 +339,7 @@ export class InstructorCourseEditPageComponent implements OnInit {
    */
   cancelEditingInstructor(index: number): void {
     const panelDetail: InstructorEditPanelDetail = this.instructorDetailPanels[index];
-    panelDetail.editPanel = this.getInstructorEditPanelModel(panelDetail.originalInstructor);
+    panelDetail.editPanel = panelDetail.originalPanel;
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts:342]: I think this is shadow copy, which means that the second time you click cancel and come again will reproduce the issue you mentioned again?

File: src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts [hunk 2]
```diff
@@ -374,6 +377,7 @@ export class InstructorCourseEditPageComponent implements OnInit {
     });
 
     panelDetail.editPanel.isEditing = false;
+    panelDetail.originalPanel = panelDetail.editPanel;
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts:380]: Why saving instructor need to restore the panel?

File: src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts [hunk 3]
```diff
@@ -438,9 +442,11 @@ export class InstructorCourseEditPageComponent implements OnInit {
         .subscribe((resp: Instructor) => {
           const newDetailPanels: InstructorEditPanelDetail = {
             originalInstructor: Object.assign({}, resp),
+            originalPanel: this.getInstructorEditPanelModel(resp),
             editPanel: this.getInstructorEditPanelModel(resp),
           };
           newDetailPanels.editPanel.permission = this.newInstructorPanel.permission;
+          newDetailPanels.originalPanel = newDetailPanels.editPanel;
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts:449]: This is shadow copy again

File: src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts (hunk 2 above)
Comment (jtankw3) [reply src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts]: It is to update the copy with the new saved values

**Pull request merged**